geoips/sector_utils/yaml_utils.py

```diff
@@ -4,10 +4,38 @@
 """Utilities for working with YAML sector specifications."""
 
 import logging
+import yaml
+import numpy as np
 
 LOG = logging.getLogger(__name__)
 
 
+# As of NumPy 2.3, NumPy scalar types (e.g., np.float64, np.int64) are no longer
+# automatically treated as native Python types by yaml.safe_dump, since numpy
+# types are no longer subclasses of native Python types as of 2.3. This code
+# explicitly registers custom representers to convert numpy types to standard Python
+# types for YAML serialization.
+def numpy_representer(dumper, data):
+    """Convert NumPy scalar types to native Python types."""
+    if isinstance(data, (np.float32, np.float64)):
+        return dumper.represent_float(float(data))
+    elif isinstance(data, (np.int32, np.int64)):
+        return dumper.represent_int(int(data))
+    elif isinstance(data, (np.bool_,)):
+        return dumper.represent_bool(bool(data))
+    elif isinstance(data, np.ndarray):
+        # Convert arrays to lists (recursive YAML representation)
+        return dumper.represent_list(data.tolist())
+    else:
+        # Fallback for other numpy types
+        return dumper.represent_str(str(data))
+
+
+# Register for SafeDumper (to ensure appropriate types in yaml.safe_dump call)
+yaml.add_multi_representer(np.generic, numpy_representer, Dumper=yaml.SafeDumper)
+yaml.add_representer(np.ndarray, numpy_representer, Dumper=yaml.SafeDumper)
+
+
```

Comment:
I'm wondering, should we be doing this at a higher level in GeoIPS? We make a LOT of use of YAML and it might be useful to ensure that this is always used, not just for sectors.